secboot/secboot_sb_test.go

```diff
@@ -1079,13 +1110,13 @@ func createMockSnapFile(snapDir, snapPath, snapType string) (snap.Container, err
 	return snapfile.Open(snapPath)
 }
 
-func mockSbTPMConnection(c *C, tpmErr error) (*sb.TPMConnection, func()) {
-	tcti, err := tpm2.OpenTPMDevice("/dev/null")
+func mockSbTPMConnection(c *C, tpmErr error) (*sb_tpm2.Connection, func()) {
+	tcti, err := linux.OpenDevice("/dev/null")
```

Comment:
This is ok, but I think it's probably a bug in go-tpm2 that linux.OpenDevice("/dev/null") opens arbitrary devices and doesn't return an error for non-TPM devices. As tpm2.NewTPMContext no longer returns an error, it's possible to call it with nil, although I think Close() would crash later on if the tests call that. It might be better in future just to call tpm2.NewTPMContext with a stub that implements the tpm2.TCTI interface (https://pkg.go.dev/github.com/canonical/go-tpm2#TCTI)

Reply:
It's bug for bug, given that in the past tpm2.OpenTPMDevice didn't return an error either, and probably should have too been mocked to implement a stub and an interface.